File: uberPaste/src/main/scala/uber/paste/openid/MD5Util.scala
package uber.paste.openid

import uber.paste.base.Loggered
import java.security.{NoSuchAlgorithmException, MessageDigest}
import java.io.UnsupportedEncodingException

/**
 * Created with IntelliJ IDEA.
 * User: achernyshev
 * Date: 13.02.13
 * Time: 20:50
 * To change this template use File | Settings | File Templates.
 */

object MD5Util {
  val instance = new MD5Util

}

class MD5Util extends Loggered{


  def hex(array:Array[Byte]):String = {

    val sb = new StringBuilder()

    for (b<-array) {
      sb.append(Integer.toHexString((b & 0xFF) | 0x100).substring(1, 3))
    }
    return sb.toString();
  }

  def md5Hex(message:String):String = {
    try {
      val md = MessageDigest.getInstance("MD5")
      return hex(md.digest(message.getBytes("UTF-8")))
    } catch{
      case e @ (_ : NoSuchAlgorithmException | _ : UnsupportedEncodingException) =>{
        logger.error(e.getLocalizedMessage,e)
        return null
      }

    }

  }


}

File: shareService/webapp/src/main/java/uber/megashare/base/MD5Util.java
/**
 * Copyright (C) 2011 alex <alex@0x08.tk>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package uber.megashare.base;

import java.io.UnsupportedEncodingException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class MD5Util extends LoggedClass {

    /**
     *
     */
    private static final long serialVersionUID = 8951807230587307847L;
    
    private static MessageDigest md;

    private MD5Util() {
        try {
            md = MessageDigest.getInstance("MD5");
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }

    public String hex(byte[] array) {
        StringBuilder sb = new StringBuilder();
        for (byte anArray : array) {
            sb.append(Integer.toHexString((anArray & 0xFF) | 0x100).substring(
                    1, 3));
        }
        return sb.toString();
    }

    public String md5Hex(String message) {
        try {
            return hex(md.digest(message.getBytes("UTF-8")));
        } catch (UnsupportedEncodingException e) {
            throw new RuntimeException(e);
        }

    }
    private static final MD5Util instance = new MD5Util();

    public static MD5Util getInstance() {
        return instance;
    }
}
